src/main/java/net/sf/jabref/gui/undo/MultipleUndoableRemoveEntries.java

```diff
@@ -0,0 +1,75 @@
+/*  Copyright (C) 2003-2016 JabRef contributors.
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.gui.undo;
+
+import java.util.List;
+import javax.swing.undo.AbstractUndoableEdit;
+import net.sf.jabref.gui.BasePanel;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.IdGenerator;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class MultipleUndoableRemoveEntries extends AbstractUndoableEdit {
+
+    private final BibDatabase base;
+    private final List<BibEntry> entries;
+    private final BasePanel panel;
+
+    private static final Log LOGGER = LogFactory.getLog(UndoableRemoveEntry.class);
+
+    public MultipleUndoableRemoveEntries(BibDatabase base, List<BibEntry> entries, BasePanel panel) {
+        this.base = base;
+        this.entries = entries;
+        this.panel = panel;
+    }
+
+
+    @Override
+    public String getUndoPresentationName() {
+        return "Undo: remove "+ (entries.size() == 1 ?  "entry" : "entries");
+    }
+
+    @Override
+    public String getRedoPresentationName() {
+        return "Redo: remove "+ (entries.size() == 1 ?  "entry" : "entries");
+    }
+
+    @Override
+    public void undo() {
+        super.undo();
+
+        entries.stream().forEach(entry -> entry.setId(IdGenerator.next()));
```

Comment:
Ultimately it shouldn't.
I took it over from `UndoableRemoveEntry`.

But the main Idea is OK?